src/halmos/sevm.py

```diff
@@ -919,43 +921,50 @@ def normalize(expr: Any) -> Any:
     def sha3(self) -> None:
         loc: int = self.st.mloc()
         size: int = int_of(self.st.pop(), "symbolic SHA3 data size")
-        self.sha3_data(wload(self.st.memory, loc, size), size)
+        self.st.push(self.sha3_data(wload(self.st.memory, loc, size), size))
 
-    def sha3_data(self, data: Bytes, size: int) -> None:
+    def sha3_data(self, data: Bytes, size: int) -> Word:
         f_sha3 = Function(
             "sha3_" + str(size * 8), BitVecSort(size * 8), BitVecSort(256)
         )
         sha3_expr = f_sha3(data)
-        sha3_output = BitVec(f"sha3_output_{len(self.sha3s):>02}", 256)
-        self.solver.add(sha3_output == sha3_expr)
+
```

Comment:
this also fails when `data` is concrete (`bytes`). Not very elegant but this wraps it up in a BV if needed:

```python
    def sha3_data(self, data: Bytes, size: int) -> Word:
        if not is_bv(data):
            if isinstance(data, bytes):
                data = int.from_bytes(data, "big")
            data = BitVecVal(data, size * 8)
```

Reply:
i'd like to minimize the mix uses of bv and bytes in the codebase.  it's not a long term solution anyway, and we need to implement a proper abstraction of symbolic values soon.

Reply:
yes agreed